library/psa_crypto.c

```diff
@@ -1227,15 +1227,16 @@ static psa_status_t psa_copy_key_material( const psa_key_slot_t *source,
     buffer_size = PSA_KEY_EXPORT_MAX_SIZE( source->type,
                                            psa_get_key_bits( source ) );
     buffer = mbedtls_calloc( 1, buffer_size );
-    if( buffer == NULL )
+    if( buffer == NULL && buffer_size != 0 )
         return( PSA_ERROR_INSUFFICIENT_MEMORY );
     status = psa_internal_export_key( source, buffer, buffer_size, &length, 0 );
     if( status != PSA_SUCCESS )
         goto exit;
     status = psa_import_key( target, source->type, buffer, length );
 
 exit:
-    mbedtls_platform_zeroize( buffer, buffer_size );
+    if( buffer_size != 0 )
+        mbedtls_platform_zeroize( buffer, buffer_size );
```

Comment:
This isn't the only place where we call `mbedtls_platform_zeroize` with a pointer that might potentially be null if the size is 0. Should we change this function to guarantee that it works in this case?

Reply:
Sounds like a separate bug outside of scope of this PR. Let's at least avoid making things worse in new code.

Reply:
Filed https://github.com/ARMmbed/mbed-crypto/issues/49